src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs

```diff
@@ -721,5 +723,37 @@ private static bool GetSupportsSha3()
 
             return false;
         }
+
+        private static bool DetermineBinaryFormatterSupport()
+        {
+            Assembly assembly = typeof(System.Runtime.Serialization.Formatters.Binary.BinaryFormatter).Assembly;
+            AssemblyName name = assembly.GetName();
+            Version assemblyVersion = name.Version;
+
+            if (IsNetFramework)
+            {
+                return true;
+            }
+
+            if (IsNativeAot)
+            {
+                Console.WriteLine("BinaryFormatter is disabled in NativeAOT");
+                return false;
+            }
+
+            bool isSupported = true;
+
+            // Version 8.1 is the version in the shared runtime (.NET 9+) that has the type disabled with no config.
+            // Assembly versions beyond 8.1 are the fully functional version from NuGet.
+            // Assembly versions before 8.1 probably won't be encountered, since that's the past.
+
+            if (assemblyVersion.Major == 8 && assemblyVersion.Minor == 1)
+            {
+                isSupported = false;
+            }
+
+            Console.WriteLine($"BinaryFormatter is from assembly version {assemblyVersion}, supported={isSupported}");
+            return isSupported;
```

Comment:
I don't recall us doing any writing to standard output like this for other tests. I am also not sure if xUnit won't redirect the output and even prevent the printing to console.

```suggestion
            if (IsNetFramework)
            {
                return true;
            }
            else if (IsNativeAot)
            {
                return false;
            }
            
            Assembly assembly = typeof(System.Runtime.Serialization.Formatters.Binary.BinaryFormatter).Assembly;
            AssemblyName name = assembly.GetName();
            Version assemblyVersion = name.Version;

            bool isSupported = true;

            // Version 8.1 is the version in the shared runtime (.NET 9+) that has the type disabled with no config.
            // Assembly versions beyond 8.1 are the fully functional version from NuGet.
            // Assembly versions before 8.1 probably won't be encountered, since that's the past.

            if (assemblyVersion.Major == 8 && assemblyVersion.Minor == 1)
            {
                isSupported = false;
            }

            return isSupported;
```

Reply:
> I am also not sure if xUnit won't redirect the output and even prevent the printing to console.

It doesn't redirect it or block it.

> I don't recall us doing any writing to standard output like this for other tests.

I know that Console.WriteLine is my go-to for debugging test failures, though they usually get removed before checkin.  I don't know offhand of anywhere that has left them in.

Since this is the second comment about removing them, I'll go ahead and do so.  They were certainly useful to me during development as it let me know if the build system was currently using the working/net8 version or the broken/net9 when I was running tests both ways.